### ISSUE TITLE
Segmentation fault in XIFreeDeviceInfo

### ISSUE THREAD

**Issue by MrMetric** [jwilm/alacritty#895]
Which operating system does the issue occur on? Arch Linux

If on linux, are you using X11 or Wayland? X11

Version: master branch, revision f68558e9dcce97e0d4d842a2d5b62e31e4124210

I ran it with `rust-gdb ./target/debug/alacritty`, waited for the crash, and got this backtrace:

```
Thread 1 "alacritty" received signal SIGSEGV, Segmentation fault.
0x00007ffff363b9fc in XIFreeDeviceInfo () from /usr/lib/libXi.so.6
(gdb) bt
#0  0x00007ffff363b9fc in XIFreeDeviceInfo () from /usr/lib/libXi.so.6
#1  0x0000555555d86967 in winit::platform::platform::x11::{{impl}}::drop (self=0x7fffffff5070)
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/platform/linux/x11/mod.rs:641
#2  0x0000555555d54395 in core::ptr::drop_in_place<winit::platform::platform::x11::DeviceInfo> () at /checkout/src/libcore/ptr.rs:60
#3  0x0000555555d86353 in winit::platform::platform::x11::EventsLoop::init_device (self=0x7fffffff8768, device=17)
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/platform/linux/x11/mod.rs:575
#4  0x0000555555745c9a in winit::platform::platform::x11::EventsLoop::process_event<&mut closure> (self=0x7fffffff8768, xev=0x7fffffff7818, callback=0x7fffffff78e0)
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/platform/linux/x11/mod.rs:553
#5  0x0000555555742313 in winit::platform::platform::x11::EventsLoop::run_forever<closure> (self=0x7fffffff8768, callback=closure = {...})
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/platform/linux/x11/mod.rs:174
#6  0x000055555574201a in winit::platform::platform::EventsLoop::run_forever<closure> (self=0x7fffffff8760, callback=closure = {...})
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/platform/linux/mod.rs:387
#7  0x0000555555741e5d in winit::EventsLoop::run_forever<closure> (self=0x7fffffff8760, callback=closure = {...})
    at /home/mrmetric/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.7.6/src/lib.rs:219
#8  0x000055555579d7ed in alacritty::window::Window::wait_events<closure> (self=0x7fffffff8760, func=closure = {...}) at /home/mrmetric/Downloads/source/alacritty/src/window.rs:274
#9  0x00005555556b0101 in alacritty::event::Processor<alacritty::event_loop::Notifier>::process_events<alacritty::event_loop::Notifier> (self=0x7fffffffc178, term=0x7fffef8d5e10, 
    window=0x7fffffff8760) at /home/mrmetric/Downloads/source/alacritty/src/event.rs:375
#10 0x00005555557a0f52 in alacritty::run (config=Config = {...}, options=Options = {...}) at src/main.rs:170
#11 0x000055555579faac in alacritty::main () at src/main.rs:45
```

I tried to do some debugging, but I don't know Rust, so I can't tell if this is a problem in Alacritty or in winit.

**Comment by francesca64** [comment]
This has been confirmed to be resolved: tomaka/winit#345

**Comment by chrisduerr** [comment]
Thanks a ton for the heads-up! As always much appreciated!
